### ISSUE TITLE
Initializer expressions / operator

### ISSUE THREAD

**Issue by LunicLynx** [dotnet/csharplang#2918]
Maybe this was considered before, but here we go!

I just realized there is an inconsistency with initializers in c#. They are always bound to the new expression but don't actually need to. (I might be wrong)

Some context:

```csharp

// an example object
public class Person
{
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public List<string> PhoneNumbers { get; } = new List<string>();
}

// field initializer
new Person {
    FirstName = "Doc",
    LastName = "Brown"
}

// List initializer
new Person {
    PhoneNumbers = {
          "01234567",
          "098765432"
    }
}

``` 

Note that the parentheses behind the type are optional.

We could write either `new Person {...}` or `new Person() {...}`, it would also work with a `ctor` that has parameters like `new Person(...) {...}`.
But the required part is the `new`, but i think this is unnecessary. The requirement should only be that before the opening brace `{` should be something invokable or an invocation.
This would allow for initialization after factory methods. Considering this setup:

```csharp

public static class PersonFactory
{
    public Person CreatePerson() => new Person();

    public Person CreatePerson(string firstName, string lastName) =>
        return new Person 
        {
            FristName = firstName,
            LastName = lastName
        };
}

// working today

// example fields
var person1 = PersonFactory.CreatePerson();
person1.FirstName = "Biff";
person1.LastName = "Tannen";

// example for lists with factory method that has parameters
var person2 = PersonFactory.CreatePerson("Marty", "McFly");
person2.PhoneNumbers.Add("0123456");

```

We currently must go to statement level if we want to do any additional initialization.
Here is what i propose, given the previous `PersonFactory`.

```csharp

// example fields
var person1 = PersonFactory.CreatePerson() {
    FirstName = "Lorraine",
    LastName = "Banes-McFly"
};
// or without parantheses
var person1 = PersonFactory.CreatePerson {
    FirstName = "George",
    LastName = "McFly"
};

// example for lists with factory method that has parameters
var person2 = PersonFactory.CreatePerson("Jennifer", "Parker") {
    PhoneNumbers = {
        "0123456"
    }
};

```

This is especially useful when used in another initializer

```csharp

public class Movie {
    public List<Person> Cast { get; } = new List<Person>();
}

new Movie {
    Cast = {
        PersonFactory.CreatePerson("Dave", "McFly") {
            PhoneNumbers = {
                "0123456"
            }
        }
    }
};

```

This would otherwise require going to statement level.

TL;DR; It should not be required to restrict initializers to be used only in correlation with the new keyword. Instead they should be allowed after something that is invokable without parameters or after invocations.

Edit:
* fix formatting

**Comment by yaakov-h** [comment]
This is a near duplicate of #803.

**Comment by LunicLynx** [comment]
@yaakov-h Thanks, i missed that one. I think this has enough distinction from the other item to leave it open for now. There is alot of talk about the `with` keyword and initializers everywhere, i want to focus on the fact that the current usage of initializers is inconsistent with other parts of the language.
Specifically the fact that: Initializer are bound to the new keyword, when in fact the only requirement is something invokable without parameters or a preceding invocation.

To be clear in other scenarios c# does not try to be so specific:
Linq syntax works whenever certain methods are present.
Current Initializers work when ever certain methods are present.

This item is only about removing the restriction of using initializers in correlation to the `new` keyword.

**Comment by setbbtjl8wwq** [comment]
@LunicLynx 
[The comment](https://github.com/dotnet/csharplang/issues/803#issuecomment-555973082) may be interested for your too.

**Comment by YairHalberstadt** [comment]
@LunicLynx I'm closing this as a duplicate of https://github.com/dotnet/csharplang/issues/803. Whilst the syntax is slightly different, and you're focusing on a different aspect, I think that it would be best to add this as a comment to that issue, since the feature is essentially the same.
